src/fides/api/ops/models/application_config.py

```diff
@@ -36,6 +39,18 @@ class ApplicationConfig(Base):
         nullable=False,
         default={},
     )  # store as encrypted JSON blob since config settings may have sensitive data
+    config_set = Column(
+        MutableDict.as_mutable(
+            StringEncryptedType(
+                JSONTypeOverride,
+                CONFIG.security.app_encryption_key,
+                AesGcmEngine,
+                "pkcs5",
+            )
+        ),
+        nullable=False,
+        default={},
+    )  # store as encrypted JSON blob since config settings may have sensitive data
```

Comment:
yeah - i think you're right to question it. my motivations were broadly around:

- trying not to assume too much about _how_ we resolve config properties at runtime, since i think there's still some question about the best way to do that. i.e. it felt like having both config records (i.e. api-set and config-set) in the db would make it simpler for us to update, moving forward, how we resolve between the two; or to allow someone to resolve in an alternative manner.
- trying to reduce dependency complexity, i.e. when thinking about the config proxy, i thought it simplified things to _just_ have it depend on the db record, to avoid potential conflicts in depending on both the `CONFIG` and the db, while the db itself depends on the `CONFIG`.

but looking back at these concerns, knowing where we've ended up with it, i don't really think storing the config-set in the db gains us a whole lot, and i think you're right to point out that it may be burdensome and a source of synchronization problems moving forward. additionally, i think it'd be easier to add this back in later if we do see a good utility for it, whereas it'd be a bit harder to take it out. so let's give ti a try!

Reply:
hey @adamsachs after this code review, I started to think more about the plus side, and I've changed my stance here!  I think your original plan was spot-on from the perspective of being able to re-use in Plus. The fact that you're storing both the api-set config and the traditionally set config in the database should make it easier to pull from one source in plus, regardless of whether that traditional config was a FidesConfig object or a FidesplusConfig object. 

And I was making a big deal about the "dangers of things getting out of sync" but that seems overblown in retrospect: it is largely just tests where we're updating the traditional config after the fact.


[Here's my thoughts on the plus side](https://ethyca.atlassian.net/wiki/spaces/EN/pages/2615083278/2023-02-10+Overriding+Plus+Config+without+Re-Deploying), but I think if both the api config and typical config are stored in the database as you describe, it should be fairly straightforward.

Reply:
ha, i guess i should trust my gut a bit more! 😄 we'll keep it as-is then, thanks for taking the time to think through this and consider how we'll extend it for your use case!

Reply:
your original design was very forward thinking with Plus!